app/controllers/v1/ControllerUtils.scala

```diff
@@ -68,6 +68,9 @@ trait ControllerUtils extends Controller with LazyLogging with Properties with I
     case ex: TimeoutException =>
       logger.error(s"TimeoutException $ex", ex.getCause)
       RequestTimeout(errAsJson(REQUEST_TIMEOUT, "request_timeout", s"This may be due to connection being blocked. $ex"))
+    case ex: Exception =>
```

Comment:
I am afraid that this is too general - issues that would have previously resulted in an ```INTERNAL_SERVER_ERROR``` will now incorrectly result in a ```NOT_FOUND```.

Note that this whole exception management block has other issues.  For example, there is a mapping from a ```ServiceUnavailableException``` to the ```SERVICE_UNAVAILABLE``` status code (503).  But the full type of that exception is ```javax.naming.ServiceUnavailableException```.  This has nothing to do with making a HTTP call i.e. this code is completely misleading and irrelevant.  If the remote service sends a 503 we will receive a response with the statusCode 503 - not some unrelated exception that happens to have a similar looking name ...

Similar code to this has been ripped out of the SBR APIs, and my suggestion would be to ignore this code.

The ```map``` in the middle of ```RestAdminDataRepository.getAdminData``` is where we should make changes.  This is given the ```WSResponse``` if we have received one.  It currently handles the 200 OK case, but treats everything else as an exception.  I suggest you add a new case here that handles the case where ```response.status == 404``` and return ```None``` (just like in the 200 case with the empty Seq.)  The existing code in ```ControllerUtils.lookupRequest``` should interpret the ```None``` as ```NOT_FOUND``` and send the appropriate status code to the client.